packages/tools/client-debugger/client-debugger/src/messaging/Utilities.ts

```diff
@@ -16,20 +17,24 @@ import { ISourcedDebuggerMessage } from "./Messages";
  *
  * @internal
  */
-export function postMessagesToWindow<TMessage extends ISourcedDebuggerMessage>(
+export function postMessagesToWindow<TMessage extends IDebuggerMessage>(
 	loggingOptions?: MessageLoggingOptions,
 	...messages: TMessage[]
 ): void {
+	const messagesWithSource: ISourcedDebuggerMessage[] = messages.map((m) => ({
+		...m,
+		source: debuggerMessageSource,
+	}));
+
 	// TODO: remove loggingOptions once things settle.
 	// If we need special logic for globalThis.postMessage maybe keep this function, but otherwise maybe remove it too.
 	if (loggingOptions !== undefined) {
 		const loggingPreamble =
 			loggingOptions?.context === undefined ? "" : `${loggingOptions.context}: `;
-		console.debug(`${loggingPreamble}Posting message to the window:`, messages);
+		console.debug(`${loggingPreamble}Posting messages to the window:`, messagesWithSource);
 	}
-
-	for (const message of messages) {
-		globalThis.postMessage?.(message, "*");
+	for (const message of messagesWithSource) {
+		globalThis.postMessage?.(message, "*"); // TODO: verify target is okay
```

Comment:
I think we can remove the comment on this line. I believe the "*" target is appropriate for this library, since it doesn't know anything about the devtools extension (or potentially other consumers).

Reply:
Addressed in #14809